substrate/frame/contracts/uapi/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+		/// allowed to use such a large amount of resources. So this value doesn't
+		/// make sense for a memory location or length.
+		const SENTINEL: u32 = u32::MAX;
+
+		(code.0 < SENTINEL).then_some(code.0)
+	}
+}
+
+impl ReturnCode {
+	/// Returns the raw underlying `u32` representation.
+	pub fn into_u32(self) -> u32 {
+		self.0
+	}
+	/// Returns the underlying `u32` converted into `bool`.
+	pub fn into_bool(self) -> bool {
+		self.0.ne(&0)
+	}
+}
+
+type Result = core::result::Result<(), Error>;
+
+#[cfg(not(feature = "std"))]
+#[inline(always)]
+fn extract_from_slice(output: &mut &mut [u8], new_len: usize) {
+	debug_assert!(new_len <= output.len());
+	let tmp = core::mem::take(output);
+	*output = &mut tmp[..new_len];
+}
+
+// is not recognizing its allocator and panic handler definitions.
+#[cfg(not(any(feature = "std", feature = "no-allocator")))]
+mod allocator;
+
+mod prelude;
+
+cfg_if::cfg_if! {
+	if #[cfg(any(feature = "ink-debug", feature = "std"))] {
+		/// Required by the `debug_print*` macros below, because there is no guarantee that
+		/// contracts will have a direct `ink_prelude` dependency. In the future we could introduce
+		/// an "umbrella" crate containing all the `ink!` crates which could also host these macros.
+		#[doc(hidden)]
+		pub use prelude::format;
+
+		/// Appends a formatted string to the `debug_message` buffer if message recording is
+		/// enabled in the contracts pallet and if the call is performed via RPC (**not** via an
+		/// extrinsic). The `debug_message` buffer will be:
+		///  - Returned to the RPC caller.
+		///  - Logged as a `debug!` message on the Substrate node, which will be printed to the
+		///    node console's `stdout` when the log level is set to `-lruntime::contracts=debug`.
+		///
+		/// # Note
+		///
+		/// This depends on the `debug_message` interface which requires the
+		/// `"pallet-contracts/unstable-interface"` feature to be enabled in the target runtime.
+		#[macro_export]
+		macro_rules! debug_print {
+			($($arg:tt)*) => ($crate::debug_message(&$crate::format!($($arg)*)));
+		}
+
+		/// Appends a formatted string to the `debug_message` buffer, as per [`debug_print`] but
+		/// with a newline appended.
+		///
+		/// # Note
+		///
+		/// This depends on the `debug_message` interface which requires the
+		/// `"pallet-contracts/unstable-interface"` feature to be enabled in the target runtime.
+		#[macro_export]
+		macro_rules! debug_println {
+			() => ($crate::debug_print!("\n"));
+			($($arg:tt)*) => (
+				$crate::debug_print!("{}\n", $crate::format!($($arg)*));
+			)
+		}
+	} else {
+		#[macro_export]
+		/// Debug messages disabled. Enable the `ink-debug` feature for contract debugging.
+		macro_rules! debug_print {
+			($($arg:tt)*) => ();
+		}
+
+		#[macro_export]
+		/// Debug messages disabled. Enable the `ink-debug` feature for contract debugging.
+		macro_rules! debug_println {
+			() => ();
+			($($arg:tt)*) => ();
+		}
+	}
+}
```

Comment:
Doesn't belong in this crate. We don't do any printing here and should not be part of the uapi. If we need to do debug printing in our fixtures we need to move it there.